File: SDKs/beezup.com/2.0/typescript/src/sdk/models/operations/getorderinvoicepreview.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOrderInvoicePreviewPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=accountId" })
  accountId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=beezUPOrderUUID" })
  beezUPOrderUUID: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=marketplaceTechnicalCode" })
  marketplaceTechnicalCode: string;
}


export class GetOrderInvoicePreviewHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Accept-Encoding" })
  acceptEncoding: string;
}


export class GetOrderInvoicePreviewRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetOrderInvoicePreviewPathParams;

  @SpeakeasyMetadata()
  headers: GetOrderInvoicePreviewHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.PreviewOrderInvoiceRequest;
}


export class GetOrderInvoicePreviewResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  beezUPCommonErrorResponseMessage?: shared.BeezUpCommonErrorResponseMessage;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  errorResponseMessage?: shared.ErrorResponseMessage;

  @SpeakeasyMetadata()
  previewOrderInvoiceResponse?: shared.PreviewOrderInvoiceResponse;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/operations/getorderinvoicepreview.py
import dataclasses
from typing import Optional
from ..shared import previeworderinvoicerequest as shared_previeworderinvoicerequest
from ..shared import beezup_common_errorresponsemessage as shared_beezup_common_errorresponsemessage
from ..shared import errorresponsemessage as shared_errorresponsemessage
from ..shared import previeworderinvoiceresponse as shared_previeworderinvoiceresponse


@dataclasses.dataclass
class GetOrderInvoicePreviewPathParams:
    account_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'accountId', 'style': 'simple', 'explode': False }})
    beez_up_order_uuid: str = dataclasses.field(metadata={'path_param': { 'field_name': 'beezUPOrderUUID', 'style': 'simple', 'explode': False }})
    marketplace_technical_code: str = dataclasses.field(metadata={'path_param': { 'field_name': 'marketplaceTechnicalCode', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetOrderInvoicePreviewHeaders:
    accept_encoding: str = dataclasses.field(metadata={'header': { 'field_name': 'Accept-Encoding', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetOrderInvoicePreviewRequest:
    headers: GetOrderInvoicePreviewHeaders = dataclasses.field()
    path_params: GetOrderInvoicePreviewPathParams = dataclasses.field()
    request: shared_previeworderinvoicerequest.PreviewOrderInvoiceRequest = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class GetOrderInvoicePreviewResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    beez_up_common_error_response_message: Optional[shared_beezup_common_errorresponsemessage.BeezUpCommonErrorResponseMessage] = dataclasses.field(default=None)
    error_response_message: Optional[shared_errorresponsemessage.ErrorResponseMessage] = dataclasses.field(default=None)
    preview_order_invoice_response: Optional[shared_previeworderinvoiceresponse.PreviewOrderInvoiceResponse] = dataclasses.field(default=None)
    
